### PR TITLE
Add info how to persist user data

### PR DESCRIPTION
**Pull request by mathbunnyru**
## Issue ticket if applicable

Fix: https://github.com/jupyter/docker-stacks/issues/1959

## Checklist (especially for first-time contributors)

- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] I will try not to use force-push to make the review process easier for reviewers
- [ ] I have updated the documentation for significant changes

### CODE REVIEW

**Comment by mathbunnyru** [comment]
@laundmo @bjornjorgensen please, take a look here.

File: README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ and want to launch a single Jupyter Application in a container.
 
 The [User Guide on ReadTheDocs](https://jupyter-docker-stacks.readthedocs.io/en/latest/) describes additional uses and features in detail.
 
-**Example 1:**
+### Example 1
```
Comment (mathbunnyru) [inline README.md:27]: This makes it possible to address it in other parts of docs.

**Comment by laundmo** [comment]
It's good to have this documented, but it doesnt address the core issue of how to persist packages installed by a user. it only documents how to persist packages installed by the admin of the container host.

**Comment by mathbunnyru** [comment]
> but it doesnt address the core issue of how to persist packages installed by a user

Could you please explain, what you mean by this?
Which packages and installed how exactly?

**Comment by laundmo** [comment]
a user of jupyter lab, who only sees the webinterface, may run `!pip install package` to install a package. These wont be persisted with the methods documented here.

**Comment by mathbunnyru** [comment]
Ok, I added a note that a package installed inside a running container will not be preserved.
Please, take a look.

**Comment by laundmo** [comment]
the core issue is that users cant interact with the docker image - by design, and because my users dont have the knowhow. knowing that it wont be persisted is good, but it shouldnt close #1959

**Comment by mathbunnyru** [comment]
> the core issue is that users cant interact with the docker image - by design

Why can't they? They can, but as it always works with docker images, some changes are lost after a container exists, and this is indeed by design.

> and because my users dont have the knowhow. 

That's why we're adding these docs.

> knowing that it wont be persisted is good, but it shouldnt close https://github.com/jupyter/docker-stacks/issues/1959

Could you make a **suggestion** what to add here?
You told this is a documentation issue and I believe I added all the neccessary information to the docs.

**Comment by laundmo** [comment]
> Why can't they? They can, but as it always works with docker images

sorry, this was unclear: they cant access the docker host or start/stop/rebuild images.

> You told this is a documentation issue

initially i thought it was. maybe its actually a feature request.

**Comment by mathbunnyru** [comment]
> > Why can't they? They can, but as it always works with docker images
> 
> sorry, this was unclear: they cant access the docker host or start/stop/rebuild images.
> 
> > You told this is a documentation issue
> 
> initially i thought it was. maybe its actually a feature request.

Ok, then please create a new issue with a feature request, which describes what you want users to be able to do (and we will discuss if it's possible).
Also, if you want to be the whole state saved, then you can use `docker commit` (you need to have access to the host machine), but this is not the best practice: https://stackoverflow.com/a/44480870/4881441

For now, I'm merging this issue as I believe there is nothing more to document on this issue.

**Comment by benz0li** [comment]
@laundmo You might be interested in my/b-data's [JupyterLab](https://github.com/b-data/jupyterlab-r-docker-stack) [docker](https://github.com/b-data/jupyterlab-python-docker-stack) [stacks](https://github.com/b-data/jupyterlab-julia-docker-stack).

Those images

1. allow to mount[^1] the whole user directory and thus preserve all user data.
2. use Python (no Conda/Mamba) and install packages in the home directory.

[^1]: Either volume or bind mount

Demo environment: https://demo.jupyter.b-data.ch

Cross references:

1. https://github.com/jupyter/docker-stacks/issues/1478#issuecomment-1062670304
2. https://github.com/jupyter/docker-stacks/issues/1969

**Pull request merged**